_posts/2020-01-30-new-age-of-bastion.md

```diff
@@ -0,0 +1,165 @@
+
+3- Improved performance by having less syscalls to kernel for yielding the time back to OS. Now we are passively looking for opportunity to execute, and yield back once as soon as possible.
+
+4- Added convenience methods for blocking pool with [spawn_blocking](https://docs.rs/bastion-executor/0.3.4/bastion_executor/blocking/fn.spawn_blocking.html) in addition to existing [spawn](https://docs.rs/bastion-executor/0.3.4/bastion_executor/pool/fn.spawn.html), [run](https://docs.rs/bastion-executor/0.3.4/bastion_executor/run/fn.run.html). You don't need to surface API to try out Bastion's Executor. You can only add `bastion-executor` to your `Cargo.toml` and you are good to go!
+
+------
+
+Bastion uses [LightProc](https://docs.rs/lightproc/0.3.4/lightproc/lightproc/index.html), and it's the crucial implementation that makes fault tolerance easier for the applications.
+
+**In LightProc we have added:**
+
+An interface for storing state! This will be used to persist messages or data in memory and can be backed by anything that you can define.
+```rust
+use lightproc::proc_stack::ProcStack;
+
+pub struct GlobalState {
+   pub amount: usize
+}
+
+ProcStack::default()
+    .with_pid(1)
+    .with_state(GlobalState { amount: 1 });
+```
+From Rust HashMap's to your NoSQL backend etc. [Wire up anything](https://docs.rs/lightproc/0.3.4/lightproc/proc_stack/struct.ProcStack.html#method.with_state) to your processes. Soon we will expose convenience APIs for mailbox preservation across restarts. [@onsails](https://github.com/onsails) is working on it [here](https://github.com/bastion-rs/bastion/pull/143).
+
+## A Big Thanks
+
+Yes, a big thanks, to everyone who enabled this project to run, supported, wrote code and make it production grade in plenty of aspects. Thanks for everything and thanks for keeping the road with us! We are getting bigger and I really like how people are passionate about what are we trying to do here! I can't thank to plenty of people who share his/her/their ideas on our Discord channel, on issues and over the PRs! Thank you all, let's continue building together! 2020 is a year that we (as Rust community) will unite over plenty of subjects, and topics. I like to close my thanks with these words (though this was a famous US Marshall Plan propaganda, this sentence pretty much sums up all of us as people working on Rust): **Whatever the weather we must move together!**
+
+## Conclusion
+In the upcoming blog posts, we will talk about how Bastion runtime is designed, which kind of design decisions are made and what are our newest initiatives. Meanwhile we are expanding our ecosystem with various projects and open to the maintainers who wants to work on distributed systems and data processing. We have pretty high aims for our projects, so come join us! Join our [Discord](https://discordapp.com/invite/DqRqtRT), look to our issues and projects at our org's GitHub!
```

Comment:
about how Bastion runtime is designed, which kind of design decisions are made and what are our newest initiatives.
about how the Bastion runtime is designed, which design decisions were made (and why) and our newest initiatives and next goals.

Meanwhile we are expanding our ecosystem with various projects and open to the maintainers who wants to work on distributed systems and data processing.
Meanwhile we are expanding our ecosystem with various projects and would love to welcome maintainers who want to work on distributed systems and data processing.

We have pretty high aims for our projects, so come join us!
We have pretty high aims for our projects, and would love to help and mentor you. Come join us!

look to our issues and projects at our org's GitHub! 
have a look at our issues and projects [our org's GitHub](https://github.com/bastion-rs)!